lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -249,7 +259,28 @@ def auth_approle(self, **kwargs):
         if secret_id is None:
             raise AnsibleError("Authentication method app role requires a secret_id")
 
-        self.client.auth_approle(role_id, secret_id)
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'approle'
```

Comment:
In my Vault environment we do not use the default mount point for the AppRole auth back end. This will allow users like me to pass in a non-default mount point.